json_serialization/reader.nim

```diff
@@ -587,28 +580,43 @@ proc readValue*[T](r: var JsonReader, value: var T)
     r.parseEnum(value)
     r.lexer.next()
 
-  elif value is SomeInteger:
+  elif value is SomeSignedInt:
     type TargetType = type(value)
-    const maxValidValue = maxAbsValue(TargetType)
+    let
+      isNegative = tok == tkNegativeInt
+      maxValidAbsValue =
+        if isNegative:
```

Comment:
Are we sure this doesn't produce overflow?

Reply:
Great question! Yes, because of the casting to `uint64`. 

`TargetType.high` will always be strictly smaller than the highest value for a `uint64`, as `TargetType` has to be a signed integer.

So after casting it to `uint64`, adding 1 to it won't produce an overflow

Reply:
Sounds gooood :), will do now!

Reply:
The `TargetType.high` will always be strictly smaller is only true on 2-s complement, btw. which, all target architectures for Status are.